### PR TITLE
Vp9 v2

### PR DESCRIPTION
**Pull request by feiwan1**

### CODE REVIEW

**Comment by feiwan1** [comment]
The failure looks like the problem of test device.

**Comment by feiwan1** [comment]
> The failure looks like the problem of test device.

All test pass now.

File: libavcodec/decode.c [hunk 1]
```diff
@@ -1134,10 +1134,12 @@ static int hwaccel_init(AVCodecContext *avctx,
...
 static void hwaccel_uninit(AVCodecContext *avctx)
 {
-    if (avctx->hwaccel && avctx->hwaccel->uninit)
-        avctx->hwaccel->uninit(avctx);
+    if (avctx->hwaccel && !(avctx->hwaccel->caps_internal & HWACCEL_CAP_RESET_WITHOUT_UNINIT)) {
+        if (avctx->hwaccel->uninit)
+            avctx->hwaccel->uninit(avctx);
 
-    av_freep(&avctx->internal->hwaccel_priv_data);
+        av_freep(&avctx->internal->hwaccel_priv_data);
```
Comment (xhaihao) [inline libavcodec/decode.c:1141]: is this freed finally ? If not, there is a memory leak.

Comment (xhaihao) [reply libavcodec/decode.c:1141]: avctx->internal->hwaccel_priv_data is freed in avcodec_close()

**Review by xhaihao — APPROVED** [review]

**Comment by feiwan1** [comment]
submitted to community:
https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7898

**Pull request closed without merging**